src/ray/gcs/store_client/redis_store_client.h

```diff
@@ -43,15 +43,76 @@ class RedisStoreClient : public StoreClient {
 
   Status AsyncGetAll(
       const std::string &table_name,
-      const SegmentedCallback<std::pair<std::string, std::string>> &callback) override;
+      const MultiItemCallback<std::pair<std::string, std::string>> &callback) override;
 
   Status AsyncDelete(const std::string &table_name, const std::string &key,
                      const StatusCallback &callback) override;
 
+  Status AsyncBatchDelete(const std::string &table_name,
+                          const std::vector<std::string> &keys,
+                          const StatusCallback &callback) override;
+
   Status AsyncDeleteByIndex(const std::string &table_name, const std::string &index_key,
                             const StatusCallback &callback) override;
 
  private:
+  /// \class RedisScanner
+  /// This class is used to scan data from Redis.
+  ///
+  /// If you called one method, should never call the other methods.
+  /// Otherwise it will disturb the status of the RedisScanner.
+  class RedisScanner {
+   public:
+    explicit RedisScanner(std::shared_ptr<RedisClient> redis_client,
+                          std::string table_name, std::string match_pattern);
+
+    Status ScanRows(
+        const MultiItemCallback<std::pair<std::string, std::string>> &callback);
```

Comment:
What about naming this `ScanKeysAndValues` and returning a map?

Reply:
fixed